### ISSUE TITLE
Difference between version 7.28.1 and 7.29.0 when doing `semantic-release publish`

### ISSUE THREAD

**Issue by tatu-aalto** [python-semantic-release/python-semantic-release#450]
## The problem
I am taking semantic release in use at first and when use  version `7.28.1` on my [project](https://github.com/F-Secure/failures-analysis)  and run: `semantic-release publish -v DEBUG --noop`, I see something like this:
```
debug: evaluate_version_bump -> minor
debug: get_new_version(0.1.0, 'minor', False)
debug: get_new_version -> 0.2.0
debug: get_repository_owner_and_name()
debug: get_repository_owner_and_name -> ('tatu-aalto', 'failures-analysis')
debug: Running publish on branch main
debug: checkout(main)
debug: checkout -> M    poetry.lock
debug: M        pyproject.toml
debug: M        utest/test_similarity.py
debug: Your branch is ahead of 'upstream/main' by 1 commit.
debug:   (use "git push" to publish your local commits)
warning: No operation mode. Should have bumped from 0.1.0 to 0.2.0
``` 
Which is what I did expect to see. But when I run same command by using `7.29.0` I see 
```
debug: Ignoring UnknownCommitMessageStyleError: Unable to parse the given commit message: Initial commit
debug: Commits found since last release: 56
debug: evaluate_version_bump -> minor
debug: get_new_version(0.1.0, '7.29.0', 'minor', False)
```` 
and then program exists with return code 1.  The full log with version `7.29.0`, can be found from GitHub [actions](https://github.com/F-Secure/failures-analysis/runs/6655295182?check_suite_focus=true).

## Expected behavior
I did `7.29.0` and `7.28.1` work in similar way. 

## Additional context
I did run the code with debugger and 7.29.0 fails on some [sanity check](https://github.com/relekang/python-semantic-release/blob/master/semantic_release/history/__init__.py#L291) and that code was added by https://github.com/relekang/python-semantic-release/pull/435 but can not figure out how it should be fixed.

**Comment by tatu-aalto** [comment]
Here is an example with successful run with [7.28.1](https://github.com/F-Secure/failures-analysis/runs/6656419720?check_suite_focus=true) version.

**Comment by vberlier** [comment]
Version `7.29.0` is also broken for me.

**Comment by mikebrant-xdelivery** [comment]
Seeing similar issue as well on recent release attempts.  It would appear that `get_new_version` is being passed 2nd parameter that is incorrect, like...

`debug: get_new_version('1.39.0', '1.16.0', None, False)`

and having error raised at code linked below

https://github.com/relekang/python-semantic-release/blob/3329eeffb077f628e4a965bc7fd922d09d6b63da/semantic_release/history/__init__.py#L286

It is unclear why the most recent release version as determined by the 2nd parameter is not being calculated correctly.

Our config is as follows:

```
[semantic_release]
branch = main
version_variable = setup.py:__version__
upload_to_pypi = false
```

We are able to perform release using `version_source = tag` to look for latest release from tag instead of commits, but this prevent automatic update of our `__version__` value.

Basically it seems something is wrong in getting last release from commits.

**Comment by mohitt** [comment]
Facing same issue

**Comment by bheijden** [comment]
[This comment](https://github.com/relekang/python-semantic-release/issues/460#issuecomment-1192261285) seems related? Could you verify that?

**Comment by tatu-aalto** [comment]
I do not anymore have an issue with the release. Most likely it is fixed in some version, therefore closing the issue.
